src/libraries/System.DirectoryServices.AccountManagement/src/System/DirectoryServices/AccountManagement/Context.cs

```diff
@@ -1014,6 +1022,8 @@ public void Dispose()
                 if (_queryCtx != null)
                     _queryCtx.Dispose();
 
+                _credValidate.Dispose();
```

Comment:
We do not need a null check here because the constructor of `PrincipalContext` always assigns an instance of `CredentialValidator` to `_credValidate` and this is the only place where `_credValidate` is modified.

Reply:
The funny bit to remember about readonly fields is that they might be null if the class has a finalizer. Namely, if the constructor throws before initializing such a field you will get a finalizer call where the field will be null.
In this case, no finalizer so it should be OK to omit a null check.